### PR TITLE
Fix concurrent HTTP connection issues

### PR DESCRIPTION
**Pull request by Uxio0**
- Add missing `timeout` for some http queries
- Increase connection pool for Coingecko, Kraken, Binance and more client APIs
- Fix typo and reduce number of retries for EthereumClient (safe-eth-py new version)

### CODE REVIEW

File: safe_transaction_service/tokens/clients/coingecko_client.py [hunk 1]
```diff
@@ -36,8 +36,10 @@ class CoingeckoClient:
     }
     base_url = "https://api.coingecko.com/"
 
-    def __init__(self, network: Optional[EthereumNetwork] = None):
-        self.http_session = requests.Session()
+    def __init__(
+        self, network: Optional[EthereumNetwork] = None, request_timeout: int = 10
```
Comment (moisses89) [inline safe_transaction_service/tokens/clients/coingecko_client.py:40]: Maybe should be `request_timeout Optional[int] = 10`, the same in other places

File: safe_transaction_service/tokens/clients/base_client.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import requests
+
+
+class BaseHTTPClient:
+    def __init__(self, request_timeout: int = 10):
```
Comment (moisses89) [inline safe_transaction_service/tokens/clients/base_client.py:5]: request_timeout is optional?

File: safe_transaction_service/tokens/clients/coingecko_client.py (hunk 1 above)
Comment (Uxio0) [reply safe_transaction_service/tokens/clients/coingecko_client.py:40]: Is not optional as it's always expecting an int

File: safe_transaction_service/tokens/clients/base_client.py (hunk 2 above)
Comment (Uxio0) [reply safe_transaction_service/tokens/clients/base_client.py:5]: Yes, and 10 seconds by default

**Pull request merged**